packages/amplify-migration/src/app_storage_definition_fetcher.ts

```
@@ -21,32 +35,88 @@ export class AppStorageDefinitionFetcher {
-      assert(await fileOrDirectoryExists(cliInputsPath));
-      const cliInputs = await this.readJsonFile(cliInputsPath);
-      const bucketName = amplifyMeta.storage[storageName].output.BucketName;
-      assert(bucketName);
-      const triggers = await this.s3Client.send(new GetBucketNotificationConfigurationCommand({ Bucket: bucketName }));
-      console.log('triggers', triggers);
-      const storageOptions = getStorageDefinition({
-        cliInputs,
-        bucketName,
-      });
-      return storageOptions;
+      for (const [storageName, storage] of Object.entries(amplifyMeta.storage)) {
+        const cliInputsPath = path.join(currentCloudBackendDirectory, 'storage', storageName, 'cli-inputs.json');
+
+        if (!(await fileOrDirectoryExists(cliInputsPath))) {
+          throw new Error(`Could not find cli-inputs.json for ${storageName}`);
+        }
+
+        const cliInputs = await this.readJsonFile(cliInputsPath);
+        const storageOutput = storage as StorageOutput;
+        if (storageOutput.service === 'S3') {
+          const bucketName = storageOutput.output.BucketName;
+          if (!bucketName) throw new Error('Could not find bucket name');
+
+          const connections = await this.s3Client.send(new GetBucketNotificationConfigurationCommand({ Bucket: bucketName }));
+          const { Status: accelerateConfiguration } = await this.s3Client.send(
+            new GetBucketAccelerateConfigurationCommand({ Bucket: bucketName }),
+          );
+          const { Status: versioningConfiguration } = await this.s3Client.send(new GetBucketVersioningCommand({ Bucket: bucketName }));
+
+          const triggers = this.getStorageTriggers(connections);
+
+          const storageDefinition = getStorageDefinition({
+            cliInputs,
+            bucketName,
+            triggers,
+          });
+
+          if (!storageOptions) storageOptions = {};
```

should we throw error when `storageOptions` is not present?

---

I believe throwing an error won't be a good option as it might be possible that no storage resource is added to the application and still we would want the codegen piece to continue. And if `'storage'` is present in amplify_meta file then storage is defined and we would want to create the `storageOptions` object.